### DISCUSSION TITLE
Help wanted! Offtop: controlling sceneitems within a group

### DISCUSSION THREAD

**Discussion by DesFelasaf** [upgradeQ/Streaming-Software-Scripting-Reference#23] [Q&A]
Offtop, help wanted!
Здравствуйте! Изучаю сейчас скриптинг для OBS, правда на пайтоне, и столкнулся с такой проблемой: не получается включать/выключать отображение объекта sceneitem, который находится внутри группы, с помощью obs_sceneitem_set_visible. Отображение самой группы легко переключается, но никак не могу достучаться до объектов внутри нее. Если не трудно, можете объяснить как правильно управлять sceneitem'ами внутри групп? Есть предположение, что группу надо понимать как отдельную сцену со своими sceneitem'ами, но не уверен что это так. Простите, что пишу здесь, не знаю больше способов как с вами связаться...

Hello! I'm learning OBS-scripting now, but it's with Python, and have faced with this issue: I can't control visibility of sceneitems that are inside of a group by obs_sceneitem_set_visible function. Visibility of the group is controlled easily, but I can't control sceneitems inside this group. If it's not difficult, could you please explain how to control sceneitems within a group? I have a suggestion that the group should be considered as a separate scene with its own sceneitems, but I'm not sure if this is the case. Sorry for creating this issue here, I don't know how to contact you by another way...

**Comment by upgradeQ** [comment]
Здравствуйте, @DesFelasaf , используйте функцию [`obs_scene_find_source_recursive`](https://obsproject.com/docs/reference-scenes.html?highlight=recursive#c.obs_scene_find_source_recursive) чтобы получить обьект предмета на сцене(группе).  Смотрите также [`#2788`](https://github.com/obsproject/obs-studio/issues/2788).

**Reply by DesFelasaf** [reply]
Спасибо! Это помогло добиться нужного эффекта) Могу задать еще один вопрос?
Мне нужно вывести объект на экран (сделать его видимым) и через пять секунд спрятать его обратно. Это нужно реализовать через timer_add? Примерно таким образом:
def callback():
    obs.obs_sceneitem_set_visible(scene_item, False)
    obs.remove_current_callback()
obs.timer_add(callback, 5000)
UPD: решил через threading
